src/main/java/com/Elessar/app/client/MyClientServer.java

```diff
@@ -25,10 +25,10 @@
     private static final Logger logger = LogManager.getLogger(MyClientServer.class);
     private final String serverName;
     private final int port;
-    private final BlockingQueue<String> messageQueue;
+    private final BlockingMsgQueue messageQueue;
     private final MetricManager metricManager;
 
-    public MyClientServer(String serverName, int port, BlockingQueue<String> messageQueue, MetricManager metricManager) {
+    public MyClientServer(String serverName, int port, BlockingMsgQueue messageQueue, MetricManager metricManager) {
```

Comment:
We probably need a stop() function in this class which simply calls 
```
server.stop();
```

And in BlockingMsgQueue's close() you would call clientServer.stop()